app/src/main/java/ar/com/wolox/android/example/ui/example/ExamplePresenter.kt

```diff
@@ -4,19 +4,48 @@ import ar.com.wolox.android.example.utils.UserSession
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
 import javax.inject.Inject
 
+fun flattenSize(m: MutableMap<String, MutableList<String>>): Int {
+    var acc = 0
+    m.forEach { (s, mutableList) -> acc += mutableList.size }
+    return acc
+}
+
 class ExamplePresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<ExampleView>() {
 
-    fun onLoginButtonClicked(user: String, color: String) {
-        userSession.username = user
-        view?.goToViewPager(color)
-    }
+    fun onLoginButtonClicked(email: String, password: String) {
+        var ErrorList: MutableMap<String, MutableList<String>> = mutableMapOf()
+
+        // TODO: How to avoid these lines ? I mean, how to create ErrorList[...] while inserting element in its list
+        ErrorList["email"] = mutableListOf()
```

Comment:
I had to write these lines because the next lines
` ... ErrorList["email"]?.add(ERROR_INPUT)`
` ... ErrorList["password"]?.add(ERROR_INPUT)`
` ... ErrorList["email"]?.add(ERROR_BAD_EMAIL)`
didn't work.

Reply:
Maybe you can create a class that fills with with this things. Maybe we can talk together about this error hadling